### ISSUE TITLE
_observableOf types misformed

### ISSUE THREAD

**Issue by wratho** [RicoSuter/NSwag#3506]
![image](https://user-images.githubusercontent.com/28812961/121263184-e183ab00-c87a-11eb-9c35-23bcb7553c00.png)

I recently updated from rxjs 6.6.7 to rxjs 7.1. I am also strictly a javascript developer and do not really understand how the NSwag product works (other coworkers set this up and helped me get this far in diagnosing the issues).

What i know:
The image posted above is the result of upgrading from rxjs 6.6.7 to 7.1. Upon research i found that NSwag was updated recently to fix rxjs 7. I updated the C# solution's package to the latest NSwag, and updated the rxjs version setting to 7.1. Regenerating the client did not fix this issue. If i remove everything in the `<>` it works.

A coworker sent me this file (https://github.com/RicoSuter/NSwag/blob/4f345e1adc9c49d8498d4018250760aa69dd8dd2/src/NSwag.CodeGeneration.TypeScript/Templates/Client.ProcessResponse.Return.liquid) with the claims that if we change `return {{ Framework.RxJs.ObservableOfMethod }}<{{ operation.ResultType }}>(<any>null);`
to: `return {{ Framework.RxJs.ObservableOfMethod }}(<any>null);` it would fix my issue.

as well 
![image](https://user-images.githubusercontent.com/28812961/121270283-054cee00-c887-11eb-9cda-8dd65ca2de89.png)
this appears to fix my issue as well. It appears that maybe the any cast here is losing the list concept (maybe)?

**Comment by dolphinsd** [comment]
Issue with upgrade from rxjs 6.6.7 to rxjs 7.1

**Comment by RicoSuter** [comment]
Can you create a PR? Would this break older versions of rxjs?

**Comment by wratho** [comment]
> Can you create a PR? Would this break older versions of rxjs?

i really am not sure about breaking older versions. Like i said in my original post i'm very much just the front end guy and am not entirely sure everything that's going on with NSwag and the API. I can see if some of my colleagues would know more about it. 

Was also more curious if that suggested change was too much when it also looks like keeping the result type where it is now, but also adding more return type to the <any>null also seemed to fix the issue. I'm not sure which direction you'd want to go.

**Comment by wratho** [comment]
With the release 13.12, and rebuilding the api with NSwag it looks like the change from #3544 may have made more changes than we intended? It doesn't appear to have any build issues with 7.X so far, but seems goofy?
![image](https://user-images.githubusercontent.com/28812961/126419842-5cb4cefb-abf3-4d55-95d3-be3d714d2920.png)

i will check with 6.X to verify this is a disaster, but if you have any ideas on what might be more correct?

**Comment by wratho** [comment]
yea on rxjs 6 the extras `[]` is bad.
![image](https://user-images.githubusercontent.com/28812961/126420327-58be6496-01d7-4ee5-a6a4-9a02b6437176.png)
the mention of testing was potentially some miscommunication. he assumed i would have placed the `[]` in all of the places this change ended up making, but instead i just tested where i expected them to go.

Me and @dhwed will attempt to make a fixed PR tomorrow, but @RicoSuter has any suggestions on how to target the places we actually want more directly we are all ears.

Thanks and sorry.

**Comment by RicoSuter** [comment]
Would this patch work for you?
https://github.com/RicoSuter/NSwag/pull/3567

**Comment by wratho** [comment]
Yea that's my coworkers PR to fix it correctly this time. This just removes the types completely from the Observable ofs, without impacting any actual returned types.

We did a much more through testing this time, so i am more confident in this one. We did test with our current configuration on 6.6.7 and 7.2, if you have any other concerns with other configuration options that our change(s) might impact, we can try to test?
